### PR TITLE
Apply retrying strategy for Ramp and Oneinch #5402

### LINKED ISSUE #5402: Sometimes Ramp isn't available until we restart app
Maybe pulling the list of tokens failed and we should retry?

### PR DESCRIPTION
**Pull request by oa-s**
Closes #5402

### CODE REVIEW

**Review by hboon — APPROVED** [review]
Thanks for the test cases :)

```
public static let defaultRetryBehavior: RetryBehavior<RunLoop> = .randomDelayed(retries: 3, delayBeforeRetry: 5, delayUpperRangeValueFrom0To: 15)
```

But can you help verify that our code for retry with delays (i.e. not this PR) work correctly, i.e. they are delayed?

**Comment by oa-s** [comment]
we already have test for that  `RetryPublisherTests`

File: AlphaWalletTests/Factories/FakeTokenSwapper.swift [hunk 1]
```diff
@@ -10,19 +10,23 @@ import Combine
...
+    var isReachablePublisher: AnyPublisher<Bool, Never> { subject.eraseToAnyPublisher() }
+    var networkBecomeReachablePublisher: AnyPublisher<Void, Never> { subject.mapToVoid().eraseToAnyPublisher() }
+
+    init(_ value: Bool) {
+        subject = .init(value)
     }
 
-    var networkBecomeReachablePublisher: AnyPublisher<Void, Never> {
-        return Just<Void>(()).eraseToAnyPublisher()
+    func set(isReachable value: Bool) {
```
[the remark this answers is not in the document]
Comment (oa-s) [reply AlphaWalletTests/Factories/FakeTokenSwapper.swift:23]: not clearly understood, could you explain?

**Comment by hboon** [comment]
> we already have test for that `RetryPublisherTests`

Thanks. Will look there.

File: AlphaWalletTests/Factories/FakeTokenSwapper.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (hboon) [reply AlphaWalletTests/Factories/FakeTokenSwapper.swift:23]: Sorry, miss this earlier. I mean:

Can we replace `func set(isReachable value: Bool) {` with using the `isReachable` property by modifying the following to include a setter?

```
var isReachable: Bool { subject.value }
```

**Comment by hboon** [comment]
@oa-s in case you were waiting for me, good to go once you are done

**Pull request merged**